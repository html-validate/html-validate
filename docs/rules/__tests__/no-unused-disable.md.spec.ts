import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<!-- [html-validate-disable-next attribute-allowed-values -- no error, disable is invalid] -->
<button type="submit"></button>`;
markup["correct-removed"] = `<!-- disable removed, no error -->
<button type="submit"></button>`;
markup["correct-error-present"] = `<!-- [html-validate-disable-next attribute-allowed-values -- element has error, disable is valid] -->
<button type="foobar"></button>`;

describe("docs/rules/no-unused-disable.md", () => {
	it("inline validation: incorrect", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"no-unused-disable":"error","attribute-allowed-values":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct-removed", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"no-unused-disable":"error","attribute-allowed-values":"error"}});
		const report = htmlvalidate.validateString(markup["correct-removed"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct-error-present", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"no-unused-disable":"error","attribute-allowed-values":"error"}});
		const report = htmlvalidate.validateString(markup["correct-error-present"]);
		expect(report.results).toMatchSnapshot();
	});
});
