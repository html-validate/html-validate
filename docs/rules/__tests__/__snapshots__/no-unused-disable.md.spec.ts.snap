// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/no-unused-disable.md inline validation: correct-error-present 1`] = `[]`;

exports[`docs/rules/no-unused-disable.md inline validation: correct-removed 1`] = `[]`;

exports[`docs/rules/no-unused-disable.md inline validation: incorrect 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 34,
        "context": {
          "ruleId": "attribute-allowed-values",
        },
        "line": 1,
        "message": ""attribute-allowed-values" rule is disabled but no error was reported",
        "offset": 33,
        "ruleId": "no-unused-disable",
        "ruleUrl": "https://html-validate.org/rules/no-unused-disable.html",
        "selector": null,
        "severity": 2,
        "size": 24,
      },
    ],
    "source": "<!-- [html-validate-disable-next attribute-allowed-values -- no error, disable is invalid] -->
<button type="submit"></button>",
    "warningCount": 0,
  },
]
`;
